akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -427,6 +428,7 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
   private lazy val shutdownHook = new Thread {
     override def run(): Unit = {
       if (!_shutdown) {
+        _shutdown = true
```

Comment:
smells like we should use `AtomicBoolean` and the the actual shutdown once and only once